glusterd/cli: enhance rebalance-status after replace/reset-brick

tshacked opened the pull request:
Rebalance status is being reset during replace/reset-brick operations.
This cause 'volume status' to shows rebalance as "not started".

Fix:
change rebalance-status to "reset due to (replace|reset)-brick"

Fixes: #1717
Change-Id: I6e3372d67355eb76c5965984a23f073289d4ff23
Signed-off-by: Tamar Shacked <tshacked@redhat.com>

tshacked committed:
glusterd/cli: enhance rebalance-status after replace/reset-brick

Rebalance status is being reset during replace/reset-brick operations.
This cause 'volume status' to shows rebalance as "not started".

Fix:
change rebalance-status to "reset due to (replace|reset)-brick"

Change-Id: I6e3372d67355eb76c5965984a23f073289d4ff23
Signed-off-by: Tamar Shacked <tshacked@redhat.com>

tshacked commented:
This is the Status displays with the current PR.
The displayed ID is the previous task ID, clearing it on glusterd side -> task doesn't pass to the CLI.
Do you think it's worth handling this on the cli side, like changing the ID to "None" in case status is being reset?

```
Task Status of Volume v3
------------------------------------------------------------------------------
Task                 : Rebalance           
ID                   : 6645fa5d-7af6-4704-9a09-a65f8243207f     
Status               : reset due to replace-brick
```

P.S. - going to add .t test later

tshacked committed:
glusterd/cli: enhance rebalance-status after replace/reset-brick

Rebalance status is being reset during replace/reset-brick operations.
This cause 'volume status' to shows rebalance as "not started".

Fix: change rebalance-status to "reset due to (replace|reset)-brick"

Fixes: #1717
Signed-off-by: Tamar Shacked <tshacked@redhat.com>

Change-Id: I1e3e373ca3b2007b5b7005b6c757fb43801fde33

mohit84 commented:
/run regression

tshacked commented:
/run regression

mohit84 commented:
/run smoke

mohit84 commented:
@tshacked smoke test is failing because you have not mentioned any bug.Kindly update issue id

nik-redhat commented:
> This is the Status displays with the current PR.
> The displayed ID is the previous task ID, clearing it on glusterd side -> task doesn't pass to the CLI.
> Do you think it's worth handling this on the cli side, like changing the ID to "None" in case status is being reset?
> 
> ```
> Task Status of Volume v3
> ------------------------------------------------------------------------------
> Task                 : Rebalance           
> ID                   : 6645fa5d-7af6-4704-9a09-a65f8243207f     
> Status               : reset due to replace-brick
> ```

I think it would be good to update the ID to none as well, or else the status message looks ambiguous/misleading to me (Just a thought)

tshacked commented:
> @tshacked smoke test is failing because you have not mentioned any bug.Kindly update issue id

I already applied it on the 2nd commit.  But the page shows the 1st commit.
Will check why the test failed..

tshacked commented:
/run smoke

nik-redhat commented:
I think the correct command is `/recheck smoke`

nik-redhat commented:
/recheck smoke

tshacked committed:
cli: changing rebal task ID to "None" in case status is being reset

Rebalance status is being reset during replace/reset-brick operations.
This cause 'volume status' to shows rebalance as "not started".

Fix:
change rebalance-status to "reset due to (replace|reset)-brick"

Fixes: #1717

Change-Id: Ia73a8bea3dcd8e51acf4faa6434c3cb0d09856d0
Signed-off-by: Tamar Shacked <tshacked@redhat.com>

schaffung commented:
/run regression

tshacked commented:
Hi @mohit84, @nik-redhat, @srijan-sivakumar 
please review the new commit which changes task id to "none" 
```
Task Status of Volume tamar
------------------------------------------------------------------------------
Task                 : Rebalance           
ID                   : None                
Status               : reset due to reset-brick
```

nik-redhat reviewed (approved):
LGTM! +1

BarakSason commented on tests/bugs/glusterd/reset-rebalance-state.t:
```diff
@@ -0,0 +1,46 @@
+. $(dirname $0)/../../volume.rc
+
+
+get_rebalance_status() {
+    $CLI volume status $V0 | egrep ^"Status   " | awk '{print $3}'
+}
+
+run_rebal_check_status() {
+    TEST $CLI volume rebalance $V0 start
+    EXPECT_WITHIN $REBALANCE_TIMEOUT "completed" rebalance_status_field $V0
+    REBAL_STATE=$(get_rebalance_status)
+    TEST [ $REBAL_STATE == "completed" ]
+}
+
+replace_brick_check_status() {
+    TEST $CLI volume replace-brick $V0 $H0:$B0/${V0}1 $H0:$B0/${V0}1_replace commit force
+    REBAL_STATE=$(get_rebalance_status)
+    TEST [ $REBAL_STATE == "reset" ]
+}
+
+reset_brick_check_status() {
+    TEST $CLI volume reset-brick $V0 $H0:$B0/${V0}2 start
+    TEST $CLI volume reset-brick $V0 $H0:$B0/${V0}2 $H0:$B0/${V0}2 commit force
+    REBAL_STATE=$(get_rebalance_status)
+    TEST [ $REBAL_STATE == "reset" ]
+}
```
Very clean test pattern. Hoping to see more like this (in glusto as well).
